### PR TITLE
Do not get bounds when extracting a DataArray

### LINKED ISSUE #144: question about time_bnds missing warning
Another question:

I am unclear about why I am getting a warning about missing time_bnds. I don't think that I need it (?)  Here is what happens:

image

image

It also happens with `cam_ts.cf.data_vars["TS"]`.

If I remove the time dimension, then I don't have a problem:
`t0 = cam_ts.isel(time=0)`
`t0.cf["TS"]`
but in many cases I need the time dimensions...

Let me know if I need to provide more information, and thanks again for the help.

### PR DESCRIPTION
**Pull request by malmans2**
Closes #144

### CODE REVIEW

File: cf_xarray/accessor.py [hunk 1]
```diff
@@ -608,10 +608,9 @@ def check_results(names, k):
 
     try:
         for name in allnames:
-            extravars = accessor.get_associated_variable_names(name)
-            # we cannot return bounds variables with scalar keys
-            if scalar_key:
-                extravars.pop("bounds")
+            extravars = accessor.get_associated_variable_names(
```
Comment (dcherian) [inline cf_xarray/accessor.py:611]: shall we instead suppress the warning here using the warnings module. There's probably a way to catch the warning and issue it later after checking `scalar_key`

Comment (malmans2) [reply cf_xarray/accessor.py:611]: Yes I think that's easy to do but we should raise 2 separate warnings. One for bounds and the other for anything else. That way we still raise a warning if for example lat/lon are missing.

Comment (malmans2) [reply cf_xarray/accessor.py:611]: https://github.com/xarray-contrib/cf-xarray/blob/48a94858111d302bdb8d72863dc58b6751283a36/cf_xarray/accessor.py#L1266-L1270

Comment (dcherian) [reply cf_xarray/accessor.py:611]: what if we added the attribute names to warning. Then we could raise the warning `if "bounds" in warning_message and scalar_key`

Comment (malmans2) [reply cf_xarray/accessor.py:611]: That would help but if we use one warning only we have to:
1. Catch the warning
2. Suppress it if only bounds are in the warning
3. Edit the warning text removing bounds if other variables are missing and issue the new warning

Comment (dcherian) [reply cf_xarray/accessor.py:611]: Oh I didn't realize (3). 

Hmm.. lets use the kwarg then but rename it to `skip_bounds`?

**Pull request merged**